### PR TITLE
Contact email

### PR DESCRIPTION
**Pull request by mjurfest**
#### Trello board reference:

* [Trello Card](https://trello.com/c/Q5ipYRWz/19-3-as-a-user-i-should-be-able-to-contact-targetmvd-administrators-so-i-can-ask-them-any-questions-i-have)

---

#### Description:

* Admin should receive an email when a user wants to contact Target

---

### CODE REVIEW

File: app/views/application_mailer/contact_admin.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div>
+  <h1>Contact admin message </h1>
+  <p>From: </p>
+  <%= @sender %>
+  <div>
+    <%= @message  %>
```
Comment (diebarral) [inline app/views/application_mailer/contact_admin.html.erb]: One space between `@message` and `%>`

File: spec/requests/api/v1/contacts/create_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+describe 'POST api/v1/contact', type: :request do
+
+  let(:contact_params) do
+    {
+      sender: Faker::Internet.email,
```
Comment (santiagovidal) [inline spec/requests/api/v1/contacts/create_spec.rb]: You don't use this in the controller, you use the current user's email.

File: spec/requests/api/v1/contacts/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  end
+  let(:user) { create(:user) }
+
+  context 'with valid params' do
+    it 'returns success' do
+      post api_v1_contact_index_path, params: contact_params, headers: auth_headers, as: :json
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'sends the email' do
```
Comment (santiagovidal) [inline spec/requests/api/v1/contacts/create_spec.rb]: Empty example?

File: spec/requests/api/v1/contacts/create_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    {
+      sender: Faker::Internet.email,
+      message: Faker::Lorem.paragraph
+    }
+  end
+  let(:user) { create(:user) }
+
+  context 'with valid params' do
+    it 'returns success' do
+      post api_v1_contact_index_path, params: contact_params, headers: auth_headers, as: :json
```
Comment (santiagovidal) [inline spec/requests/api/v1/contacts/create_spec.rb]: This route is wrong. It's not the index.

File: spec/requests/api/v1/contacts/create_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+describe 'POST api/v1/contact', type: :request do
+
+  let(:user) { create(:user) }
+  let(:contact_params) do
+    {
+      sender: user.email,
```
Comment (santiagovidal) [inline spec/requests/api/v1/contacts/create_spec.rb]: You can remove this.

File: spec/requests/api/v1/contacts/create_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      message: Faker::Lorem.paragraph
+    }
+  end
+
+  context 'with valid params' do
```
Comment (santiagovidal) [inline spec/requests/api/v1/contacts/create_spec.rb]: It would be good to also test that an email is sent, since that is the purpose of this feature.

**Pull request merged**